src/Uno.Extensions.Core/IDispatcher.cs

```diff
@@ -0,0 +1,6 @@
+﻿namespace Uno.Extensions;
+
+public interface IDispatcher
```

Comment:
For perf consideration, I would add `void Enqueue(Action action, CancellationToken ct)` on the interface. This will avoid to create a `TaskCompletionSource` when not needed. 

Maybe we should also add a `void Enqueue(ActionAsync action, CancellationToken ct)` which would add a `try/catch` around the action an log exception to prevent silent/crashy exceptions in `async void`.

Both could be flagged with the `[EditorBrowsable(EditorBrowsableState.Advanced)]` attribute.

Reply:
I've pushed these suggestions to a new issue #385